tests/dataframe/test_sample.py

```
@@ -0,0 +1,88 @@
+from __future__ import annotations
+
+import pytest
+
+
+def test_sample_fraction(make_df, valid_data: list[dict[str, float]]) -> None:
+    df = make_df(valid_data)
+
+    df = df.sample(fraction=0.5)
+    df.collect()
+
+    assert len(df) == 2
+    assert df.column_names == list(valid_data[0].keys())
+
+
+def test_sample_negative_fraction(make_df, valid_data: list[dict[str, float]]) -> None:
+    df = make_df(valid_data)
+
+    df = df.sample(fraction=-1.0)
+
+    with pytest.raises(ValueError, match="negative fraction"):
+        df.collect()
+
+
+def test_sample_full_fraction(make_df, valid_data: list[dict[str, float]]) -> None:
+    df = make_df(valid_data)
+
+    df = df.sample(fraction=1.0)
+    df.collect()
+
+    assert len(df) == len(valid_data)
+    assert df.column_names == list(valid_data[0].keys())
+
+
+def test_sample_over_sample(make_df, valid_data: list[dict[str, float]]) -> None:
+    df = make_df(valid_data)
+
+    df = df.sample(fraction=2.0)
```

we should throw an error if the fraction is over 1.0